### PR TITLE
When no location, defaults to cal video

### LINKED ISSUE #2900: When no location is set in an event, where would the meeting occur ?
Found a bug? Please fill out the sections below. 👍

### Issue Summary
When no location is selected in an event, the email received after booking has a `Where` heading but no text. Maybe we should make it very clear that no location was selected by the organizer and the booker should contact the organizer for the meeting location.

Also, maybe we should not allow empty locations at all. If the organizer chooses to set no location, he should be asked to drop a comment about where the meeting is supposed to happen. 

This is to solve the confusion for a booker when such an event is booked.

Screenshot 2022-05-27 at 11 10 55 AM

A summary of the issue. This needs to be a clear detailed-rich summary.

### Steps to Reproduce

1. Create an event. By default it has Cal Video as location.
2. Go and remove that location.
3. Book a meeting
4. Check the email recieved.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details

- Browser version, screen recording, console logs, network requests: You can make a recording with [Bird Eats Bug](https://birdeatsbug.com/).
- Node.js version
- Anything else that you think could be an issue.

### PR DESCRIPTION
**Pull request by om-ray**
## What does this PR do?

Currently, if an event type does not have a location set for it, the email that is sent will have an empty `Where` heading with no text that tells you what platform the meeting will be on. This PR solves that issue by defaulting to Cal video instead of just having no location set.

Fixes #2900 

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by PeerRich** [comment]
we'll review today!

**Comment by om-ray** [comment]
@PeerRich Thank you!

File: packages/app-store/exchange2013calendar/lib/CalendarService.ts [hunk 1]
```diff
@@ -33,6 +33,8 @@ import {
   NewCalendarEventType,
 } from "@calcom/types/Calendar";
 
+import { LocationType } from "../../locations";
```
Comment (zomars) [inline packages/app-store/exchange2013calendar/lib/CalendarService.ts:36]: This type is not being used

File: packages/app-store/exchange2016calendar/lib/CalendarService.ts [hunk 2]
```diff
@@ -33,6 +33,8 @@ import {
   NewCalendarEventType,
 } from "@calcom/types/Calendar";
 
+import { LocationType } from "../../locations";
```
Comment (zomars) [inline packages/app-store/exchange2016calendar/lib/CalendarService.ts:36]:
```suggestion
```

File: packages/app-store/exchange2013calendar/lib/CalendarService.ts (hunk 1 above)
Comment (zomars) [inline packages/app-store/exchange2013calendar/lib/CalendarService.ts:36]:
```suggestion
```

**Review by zomars — CHANGES_REQUESTED** [review]
Thank you for your contribution 🙏 Let's leave the `yarn.lock` out of the PR

**Comment by zomars** [comment]
@om-ray The lock file is still present

**Comment by om-ray** [comment]
@zomars when i remove the lockfile the checks fail, but ill delete it if thats fine

**Comment by zomars** [comment]
@om-ray I don't mean to delete the file. I mean to undo the changes and leave it the same as the `main` branch.

**Comment by om-ray** [comment]
@zomars ahhh that makes more sense, ill do that right now

**Comment by om-ray** [comment]
@zomars had a little trouble with it, but its gone now!

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 3]
```diff
@@ -68,7 +69,8 @@ export const bookingsRouter = createProtectedRouter()
       newLocation: z.string(),
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/bookings.tsx:69]:
```suggestion
      newLocation: z.string().transform((val) => val || LocationType.Daily),
```

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 4]
```diff
@@ -68,7 +69,8 @@ export const bookingsRouter = createProtectedRouter()
...
     }),
     async resolve({ ctx, input }) {
-      const { bookingId, newLocation: location } = input;
+      const { bookingId, newLocation: location } =
+      input.newLocation === "" ? { bookingId: input.bookingId, newLocation: LocationType.Daily } : input;
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/bookings.tsx:73]:
```suggestion
      const { bookingId, newLocation: location } = input;
```

**Review by zomars — CHANGES_REQUESTED** [review]
BTW we can achieve the same behavior by using a zod transform

**Review by zomars — APPROVED** [review]
Thank you for your contribution 🙏

**Pull request merged**